webviz_ert/controllers/multi_response_controller.py

```diff
@@ -196,4 +202,32 @@ def _generate_plot(ensemble_id: str, color: str) -> Optional[ResponsePlotModel]:
             for observation in plot._observations:
                 fig.add_trace(observation.repr)
         fig.update_layout(assets.ERTSTYLE["figure"]["layout"])
+        fig.update_layout({"xaxis": {"title": {"text": x_axis_label}}})
+        fig.update_layout(assets.ERTSTYLE["figure"]["layout-value-y-axis-label"])
         return fig
+
+
+def axis_label_for_ensemble_response(
+    ensemble: "EnsembleModel", responseName: str
+) -> str:
+    response = ensemble.responses[responseName]
+    response_data = response.data
+    index = response_data.index
+    index_type = type(index)
+    data_sample = index[0]
+    if index_type == pd.core.indexes.base.Index:
+        # try to convert to date-time
+        try:
+            pd.to_datetime(data_sample)
+        except Exception:
+            pass
+        else:
+            return "Date"
+    if index_type == pd.core.indexes.numeric.Int64Index:
+        return "Index"
+
+    logger.warning(
+        f"got type `{index_type}` and data sample `{data_sample}` - not"
```

Comment:
Maybe add some more information to this warning to help us figure out what is happening. I.e i would guess it would be beneficial having response name / ensemble name available.

Reply:
done and done.

should I add context as well, like that we're trying to figure out an axis label based on the index? prepend with the function name? or is that information given by the logger?
or it perhaps too much information? or is one supposed to find this info by just searching in the code, if one is interested?